nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/sensitive/StandardSensitivePropertyProvider.java

```diff
@@ -0,0 +1,91 @@
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.security.Security;
+
+/**
+ *
+ * This class describes the available sensitive property providers {@link SensitivePropertyProvider} which can be chosen to encrypt/decrypt the sensitive properties in the nifi.properties file.
+ * Most of these providers will require external setup in their respective system (eg. Hashicorp Vault) and then configuration given in the nifi.properties file.
+ *
+ * For further details, review the <a href="https://nifi.apache.org/docs/nifi-docs/html/user-guide.html#sensitive_property_providers">Apache NiFi User Guide - Sensitive Property Providers</a> section.
+ *
+ * As of Apache NiFi 1.12.0 this implementation is considered <a href="https://nifi.apache.org/docs/nifi-docs/html/user-guide.html#experimental_warning">*experimental*</a>.
+ * Available implementations can be found listed in StandardSensitivePropertyProvider.
+ *
+ * This class hides the various SPP subclass construction from clients.
+ *
+ */
+public class StandardSensitivePropertyProvider {
+    private static final Logger logger = LoggerFactory.getLogger(StandardSensitivePropertyProvider.class);
+
+    static {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    /**
+     * Creates a {@link SensitivePropertyProvider} suitable for a given key or key id.
+     *
+     * If an empty or null key/key id is given, this implementation returns null.  This is a convenience
+     * for clients using the various Property classes, as those classes allow a null SensitivePropertyProvider.
+     *
+     * If no provider recognizes a key/key id, this implementation throws {@link SensitivePropertyProtectionException}.
+     *
+     * @param key provider encryption key
+     * @return concrete instance of SensitivePropertyProvider, or null when no key/key id is specified
+     * @throws SensitivePropertyProtectionException when a key/key id is not handled by any provider.
+     */
+    public static SensitivePropertyProvider fromKey(String key) {
+        if (StringUtils.isBlank(key)) {
+            return null;
+        } else if (KeyStoreWrappedSensitivePropertyProvider.isProviderFor(key)) {
+            logger.debug("StandardSensitivePropertyProvider selected specific KeyStore provider for key: " + KeyStoreWrappedSensitivePropertyProvider.toPrintableString(key));
+            return new KeyStoreWrappedSensitivePropertyProvider(key);
+        } else if (AESSensitivePropertyProvider.isProviderFor(key)) {
+            logger.debug("StandardSensitivePropertyProvider selected specific AES provider for key: " + AESSensitivePropertyProvider.toPrintableString(key));
+            return new AESSensitivePropertyProvider(key);
+        }
+
+        throw new SensitivePropertyProtectionException("No sensitive property provider for key or key id.");
```

Comment:
There are a handful of exception messages that could benefit from telling the user more. Can we make this one something like `String.format("No sensitive property provider for key or key id: %s", key);` if it's `keystore/` key id? (Assuming we don't want to dump an actual key into the logs.)